cpp/source/deal.II/electrochemical_physics.templates.h

```diff
@@ -0,0 +1,257 @@
+/* Copyright (c) 2016, the Cap authors.
+ *
+ * This file is subject to the Modified BSD License and may not be distributed
+ * without copyright and license information. Please refer to the file LICENSE
+ * for the text and further information on this license.
+ */
+
+#ifndef CAP_DEAL_II_ELECTROCHEMICAL_PHYSICS_TEMPLATES_H
+#define CAP_DEAL_II_ELECTROCHEMICAL_PHYSICS_TEMPLATES_H
+
+#include <cap/deal.II/electrochemical_physics.h>
+#include <boost/assert.hpp>
+#include <deal.II/base/function.h>
+#include <deal.II/dofs/dof_tools.h>
+#include <deal.II/fe/fe_values.h>
+#include <deal.II/numerics/vector_tools.h>
+
+namespace cap
+{
+template <int dim>
+ElectrochemicalPhysics<dim>::ElectrochemicalPhysics(
+    std::shared_ptr<PhysicsParameters<dim> const> parameters)
+    : Physics<dim>(parameters)
+{
+  std::shared_ptr<boost::property_tree::ptree const> database =
+      parameters->database;
+
+  // clang-format off
+  this->solid_potential_component  = database->get<unsigned int>("solid_potential_component");
+  this->liquid_potential_component = database->get<unsigned int>("liquid_potential_component");
+  this->temperature_component      = database->get<unsigned int>("temperature_component");
+  // clang-format on
+
+  alpha             = database->get<double>("material_properties.alpha", 0.0);
```

Comment:
Why do you have ``alpha`` here?
I used it for the reversible term in the heat generation...